crates/uv-auth/src/keyring.rs

```diff
@@ -65,7 +66,7 @@ impl KeyringProvider {
         //      use-cases.
         let key = (host.to_string(), username.to_string());
         {
-            let cache = self.cache.lock().unwrap();
+            let cache = self.cache.read().await;
```

Comment:
Once again, we can avoid holding this lock across await points — should it be sync?

Reply:
Yeah if you don't need to hold it over an await point, I think we should default to synchronous primitives.

Reply:
We could also take the write lock immediately, since we take it later? Idk if you should take read  then take write or take write once.

Reply:
I would just do the simplest thing. I'm not sure what happens if you take a read lock while also holding a write lock though. (Taking a write lock while holding a read lock is I think a deadlock? That is, std provides no way to "upgrade" a read lock to a write lock.)

Reply:
> Yeah if you don't need to hold it over an await point, I think we should default to synchronous primitives.

+1，an asynchronous rwlock doesn't seem necessary here, quote from tokio document
> an asynchronous mutex is more expensive than an ordinary mutex